File: backend/src/main/java/com/xxgwy/common/exception/BizException.java
package com.xxgwy.common.exception;

import lombok.Getter;

@Getter
public class BizException extends RuntimeException {
    private final int code;

    public BizException(int code, String message) {
        super(message);
        this.code = code;
    }

    public BizException(String message) {
        super(message);
        this.code = 500;
    }

    public static BizException unauthorized(String msg) {
        return new BizException(401, msg);
    }

    public static BizException forbidden(String msg) {
        return new BizException(403, msg);
    }

    public static BizException notFound(String msg) {
        return new BizException(404, msg);
    }

    public static BizException conflict(String msg) {
        return new BizException(409, msg);
    }

    public static BizException badRequest(String msg) {
        return new BizException(400, msg);
    }
}

File: backend/src/main/kotlin/com/xxgwy/common/exception/BizException.kt
package com.xxgwy.common.exception

class BizException(val httpStatus: Int, message: String) : RuntimeException(message) {

    companion object {
        fun conflict(message: String): BizException = BizException(409, message)
        fun unauthorized(message: String): BizException = BizException(401, message)
        fun notFound(message: String): BizException = BizException(404, message)
        fun badRequest(message: String): BizException = BizException(400, message)
        fun forbidden(message: String): BizException = BizException(403, message)
    }
}
